docs/content/Contribution guidelines/operations/code-management.md

```diff
@@ -7,3 +7,128 @@
 <!-- Code management
   Prow, Gh actions broken links, pr verifier, emoji in titles of prs, add issue to project. Add pr to project. Check spelling errors, wordlist.txt, 
 Quay.io -->
+# Code Management
+
+## Fork kubestellar into your own repo, create a local branch, set upstream to kubestellar, add and commit changes to local branch, and squash your commits
+
+### Fork the Github kubestellar repo into your own Github repo:
+You can do this either 1: from the kubestellar Github website using the "Fork" button or 2: by using the git fork command from your local git command line interface, such as git bash.
+
+copy the forked repo from Github to your local system by using the "git clone" command or by downloading the repository's zip file.
+
+In your new local forked repo, set upstream to kubestellar main
+
+check what your repo's remote settings are
+```
+git remote -v
+```
+
+### Set upstream to use kubestellar: 
+```
+git remote add upstream git@github.com:kubestellar/kubestellar.git
+```
+
+For example:
+```
+owner@BOOK-U0EMIUAFHD MINGW64 ~/src/edge-mc (main)
+ git remote -v
+origin  git@github.com:fileppb/edge-mc.git (fetch)
+origin  git@github.com:fileppb/edge-mc.git (push)
+
+owner@BOOK-U0EMIUAFHD MINGW64 ~/src/edge-mc (main)
+ git remote add upstream git@github.com:kubestellar/kubestellar.git
+
+owner@BOOK-U0EMIUAFHD MINGW64 ~/src/edge-mc (main)
+ git remote -v
+origin  git@github.com:fileppb/edge-mc.git (fetch)
+origin  git@github.com:fileppb/edge-mc.git (push)
+upstream        git@github.com:kubestellar/kubestellar.git (fetch)
+upstream        git@github.com:kubestellar/kubestellar.git (push)
+
+owner@BOOK-U0EMIUAFHD MINGW64 ~/src/edge-mc (main)
+ git fetch upstream
+Enter passphrase for key '/c/Users/owner/.ssh/id_rsa':
+remote: Enumerating objects: 60394, done.
+remote: Counting objects: 100% (5568/5568), done.
+remote: Compressing objects: 100% (255/255), done.
+remote: Total 60394 (delta 4768), reused 5457 (delta 4706), pack-reused 54826
+Receiving objects: 100% (60394/60394), 52.38 MiB | 3.25 MiB/s, done.
+Resolving deltas: 100% (34496/34496), completed with 415 local objects.
+
+owner@BOOK-U0EMIUAFHD MINGW64 ~/src/edge-mc (main)
+```
+ git status
+
+On branch main
+Your branch is up to date with 'origin/main'.
+
+nothing to commit, working tree clean
+```
+```
+### Select an issue to work on and create a local branch, 
+
+Create a local branch for your work, preferably including the issue number in the branch name
+
+for example if working on issue #11187, then you might name your local branch "issue-1187"
+```
+git checkout -b issue-1187
+```
+
+### As you work and change files, you should try to commit relatively small pieces of work, using the following commands
+```
+git add (there are several options you can specify for the git add command)
+
+git commit -m "your message"
+
+git push -u origin branch-name (-u sets upstream to origin which is your remote github repository)
```

Comment:
You only need the `-u` the first time you push a given branch.

Reply:
Yes, but I was tried to keep the verbiage to a minimum. The -u doesn't hurt on subsequent pushes.